src/main/java/com/teragrep/nlf_01/types/ADFPipelineRunType.java

```diff
@@ -0,0 +1,190 @@
+
+import com.teragrep.akv_01.event.ParsedEvent;
+import com.teragrep.akv_01.plugin.PluginException;
+import com.teragrep.nlf_01.PropertiesJson;
+import com.teragrep.nlf_01.util.ASCIIString;
+import com.teragrep.nlf_01.util.MD5Hash;
+import com.teragrep.nlf_01.util.ResourceId;
+import com.teragrep.nlf_01.util.ValidRFC5424AppName;
+import com.teragrep.nlf_01.util.ValidRFC5424Hostname;
+import com.teragrep.nlf_01.util.ValidRFC5424Timestamp;
+import com.teragrep.rlo_14.Facility;
+import com.teragrep.rlo_14.SDElement;
+import com.teragrep.rlo_14.Severity;
+import jakarta.json.JsonObject;
+import jakarta.json.JsonValue;
+import java.time.Instant;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.UUID;
+
+public final class ADFPipelineRunType implements EventType {
+
+    private final ParsedEvent parsedEvent;
+    private final String realHostname;
+
+    public ADFPipelineRunType(final ParsedEvent parsedEvent, final String realHostname) {
+        this.parsedEvent = parsedEvent;
+        this.realHostname = realHostname;
+    }
+
+    private void assertKey(final JsonObject obj, final String key, final JsonValue.ValueType type)
+            throws PluginException {
+        if (!obj.containsKey(key)) {
+            throw new PluginException(new IllegalArgumentException("Key " + key + " does not exist"));
+        }
+
+        if (!obj.get(key).getValueType().equals(type)) {
+            throw new PluginException(new IllegalArgumentException("Key " + key + " is not of type " + type));
+        }
+    }
+
+    @Override
+    public Severity severity() {
+        return Severity.NOTICE;
+    }
+
+    @Override
+    public Facility facility() {
+        return Facility.AUDIT;
+    }
+
+    @Override
+    public String hostname() throws PluginException {
+        final JsonObject record = parsedEvent.asJsonStructure().asJsonObject();
+
+        assertKey(record, "_ResourceId", JsonValue.ValueType.STRING);
+        final String resourceId = record.getString("_ResourceId");
+
+        return new ValidRFC5424Hostname(
+                "md5-".concat(new MD5Hash(resourceId).md5().concat("-").concat(new ASCIIString(new ResourceId(resourceId).resourceName()).withNonAsciiCharsRemoved()))
+        ).hostnameWithInvalidCharsRemoved();
+
+    }
+
+    @Override
+    public String appName() throws PluginException {
+        final JsonObject record = parsedEvent.asJsonStructure().asJsonObject();
+
+        assertKey(record, "PipelineName", JsonValue.ValueType.STRING);
+
+        return new ValidRFC5424AppName(record.getString("PipelineName")).validAppName();
+
+    }
+
+    @Override
+    public long timestamp() throws PluginException {
+        final JsonObject record = parsedEvent.asJsonStructure().asJsonObject();
+
+        assertKey(record, "TimeGenerated", JsonValue.ValueType.STRING);
+
+        return new ValidRFC5424Timestamp(record.getString("TimeGenerated")).validTimestamp();
+    }
+
+    @Override
+    public Set<SDElement> sdElements() {
```

Comment:
is this shared exactly in each type? perhaps we should have an object for this as well such as DefaultTypeSDE or so. @eemhu comments please and issue if relevant.